### PR TITLE
fix issue 10444 - writeln of a SIMD register

### PR DESCRIPTION
**Pull request by BBasile**
In 2014 there was two previous attempts:
- https://github.com/dlang/phobos/pull/2124: never merged because it required a rebase which was never done, also troubles with the tests.
- https://github.com/dlang/phobos/pull/2220: was closed because the other PR existed already.

Then it was completely forgotten for 3 years for no reason.

### CODE REVIEW

File: std/format.d [hunk 1]
```diff
@@ -3906,6 +3906,29 @@ if (isPointer!T && !is(T == enum) && !hasToString!(T, Char))
...
+void formatValue(Writer, V, Char)(auto ref Writer w, auto ref V val, const ref FormatSpec!Char f)
+if (isSIMDVector!V)
+{
+    formatValue(w, val.array, f);
+}
+
+@safe pure unittest
+{
+    import core.simd;
+    static if (is(int4))
```
Comment (ZombineDev) [inline std/format.d]: Use `float4` is it is available on all platforms supporting SIMD instructions. Integer SIMD support is much more rare.

**Comment by BBasile** [comment]
I've amended at least 3 times and OSX 32 bit still and always fails, does someone have an idea why ?

File: std/format.d [hunk 2]
```diff
@@ -3906,6 +3906,43 @@ if (isPointer!T && !is(T == enum) && !hasToString!(T, Char))
...
+unittest
+{
+    version (OSX)
+    {
+        version (X86)
+        {
+            import core.simd;
+            static if (is(float4))
+            {
+                //float4 f;
```
Comment (BBasile) [inline std/format.d]: After running remote code for 3 hours it appears that there's a codegen error or something, see https://issues.dlang.org/show_bug.cgi?id=17823.

When the declaration of `f` is commented the test for OSX 32 bits doesn't crash anymore.

Comment (MetaLang) [reply std/format.d]: Even with this commented out it's segfaulting on CircleCI.

Comment (BBasile) [reply std/format.d]: @ZombineDev: can you mark it as blocked please ? Also remove the polution from CodeCov

**Comment by BBasile** [comment]
The tests pass but the OSX 32 bits issue is avoided with versions. I don't know why do you still test it since it's even not officially distributed.

**Comment by JackStouffer** [comment]
>I don't know why do you still test it since it's even not officially distributed.

IMO OSX 32 support should be dropped post haste. There's tons of extra work for the DMD folks to support a dead platform. For some reason Walter wants to keep it.

**Pull request merged**